kuma/search/utils.py

```diff
@@ -72,13 +75,17 @@ def search_exception_handler(exc):
     # to get the standard error response.
     response = exception_handler(exc)
 
-    if (response is None and
-            isinstance(exc, elasticsearch.ElasticsearchException)):
-        # FIXME: This really should return a 503 error instead but Zeus
-        # doesn't let that through and displays a generic error page in that
-        # case which we don't want here
-        log.error('Elasticsearch exception: %s' % exc)
-        return Response({'error': SEARCH_DOWN_DETAIL},
-                        status=status.HTTP_200_OK)
+    if response is None:
+        if isinstance(exc, elasticsearch.ElasticsearchException):
+            # FIXME: This really should return a 503 error instead but Zeus
+            # doesn't let that through and displays a generic error page in that
+            # case which we don't want here
+            log.error('Elasticsearch exception: %s' % exc)
+            return Response({'detail': SEARCH_DOWN_DETAIL},
+                            status=status.HTTP_200_OK)
+        elif isinstance(exc, UnicodeDecodeError):
+            log.error('UnicodeDecodeError exception: %s' % exc)
+            return Response({'detail': SEARCH_ERROR_DETAIL},
+                            status=status.HTTP_404_NOT_FOUND)
```

Comment:
A plain 400 is probably more appropriate for these kinds of errors.

Also, this path isn't reached via [the url originally posted in the bug](https://developer-local.allizom.org/en-US/search?q=%BF%27%22%28&topic=api&skill=intermediate&type=howto%2F), so I assume this is defensive coding in case something else raises a `UnicodeDecodeError`?